src/unit_tests/wazuh_db/test_wdb_global_helpers.c

```diff
@@ -3635,12 +3535,377 @@ void test_wdb_parse_chunk_to_int_err(void **state) {
+    will_return(__wrap_cJSON_AddItemToArray, true);
+    expect_string(__wrap_cJSON_AddNumberToObject, name, "id");
+    expect_value(__wrap_cJSON_AddNumberToObject, number, 1);
+    will_return_always(__wrap_cJSON_AddNumberToObject, 1);
+
+    // Json array items loop
+    expect_string(__wrap_cJSON_CreateString, string, "default");
+    will_return(__wrap_cJSON_CreateString, 1);
+    expect_function_call(__wrap_cJSON_AddItemToArray);
+    will_return(__wrap_cJSON_AddItemToArray, true);
+    expect_string(__wrap_cJSON_CreateString, string, "Group1");
+    will_return(__wrap_cJSON_CreateString, 1);
+    expect_function_call(__wrap_cJSON_AddItemToArray);
+    will_return(__wrap_cJSON_AddItemToArray, true);
+    expect_string(__wrap_cJSON_CreateString, string, "Group2");
+    will_return(__wrap_cJSON_CreateString, 1);
+    expect_function_call(__wrap_cJSON_AddItemToArray);
+    will_return(__wrap_cJSON_AddItemToArray, true);
+
+    // Printing JSON
+    will_return(__wrap_cJSON_PrintUnformatted, data_in_str);
+    expect_function_call(__wrap_cJSON_Delete);
+
+    // Calling Wazuh DB
+    expect_value(__wrap_wdbc_query_ex, *sock, socket);
+    expect_string(__wrap_wdbc_query_ex, query, query_str);
+    expect_value(__wrap_wdbc_query_ex, len, WDBOUTPUT_SIZE);
+    will_return(__wrap_wdbc_query_ex, response);
+    will_return(__wrap_wdbc_query_ex, OS_SUCCESS);
+
+    // Parsing Wazuh DB result
+    expect_any(__wrap_wdbc_parse_result, result);
+    will_return(__wrap_wdbc_parse_result, WDBC_OK);
+
+    res = wdb_set_agent_groups_csv(id, groups_csv, mode, sync_status, &socket);
+
+    assert_int_equal(OS_SUCCESS,res);
+
+    free_strarray(groups_array);
+    os_free(groups_csv);
+    os_free(mode);
+    os_free(sync_status);
+    os_free(query_str);
+    os_free(response);
+    __real_cJSON_Delete(j_data_in);
+    __real_cJSON_Delete(j_agent_info);
+}
+
+void test_wdb_set_agent_groups_error_no_mode(void **state) {
+    char** groups_array = NULL;
+    char* mode = NULL;
+    char* sync_status = NULL;
+    int id = 1;
+    int socket = -1;
+    int res;
+
+    os_calloc(4, sizeof(char *), groups_array);
+    os_strdup("default", groups_array[0]);
+    os_strdup("Group1", groups_array[1]);
+    os_strdup("Group2", groups_array[2]);
+    groups_array[3] = NULL;
+    os_strdup("synced", sync_status);
+
+    // Debug message
+    expect_string(__wrap__mdebug1, formatted_msg, "Invalid params to set the agent groups 01");
+
+    res = wdb_set_agent_groups(id, groups_array, mode, sync_status, &socket);
+
+    assert_int_equal(OS_INVALID,res);
+
+    free_strarray(groups_array);
+    os_free(sync_status);
+}
+
+void test_wdb_set_agent_groups_socket_error(void **state) {
+    char** groups_array = NULL;
+    char* mode = NULL;
+    char* sync_status = NULL;
+    char* query_str = NULL;
+    char* data_in_str = NULL;
+    char* response = NULL;
+    int id = 1;
+    int socket = -1;
+    int res;
+    cJSON* j_data_in = __real_cJSON_CreateObject();
```

Comment:
all of the `cJSON_AddXX` functions are mocked so we are not really adding items to a json object so i don't think we really need to create a real json object. It should be sufficient to return a non-null value from `__wrap_cJSON_CreateObject`